src/Plans.jl

```diff
@@ -632,25 +663,82 @@ Display a comparison of two plans, or save it in a text file.
+compare_plans(io::Union{IOBuffer, Base.TTY}, left::Plan, right::Plan; kwargs...) = compare_plans(left, right; io=io, kwargs...)
+
+function compare_plans(left::Plan, right::Plan; outfile = "",
+        io = nothing, 
+        alphabetical=false,  # whether to sort variables
+        pagelines=0,
+        exog_mark="X", endo_mark="~", missing_mark=".", # symbols used for each class
+        delim=" ",  # set to "," to get a CSV file (with 3 skip rows and 1 header row)
+        _name_delim=delim,  # padding after NAME column
+        _range_delim=delim,    # padding between range columns
+        diff = false, # whether to print only variables/shocks which differ
+        summary = false, # whether to print summary information to stdout
+        legend = false # whether to print the legend for the matrix numbers
+    ) #
+    left_vars = keys(left.varshks)
+    right_vars = keys(right.varshks)
+    aligned_right = deepcopy(left)
+    aligned_right.exogenous .= 0
+    copyto!(aligned_right, rangeof(left), right; verbose=false)
+    left_matrix = Int64.(left.exogenous)
+    right_matrix = Int64.(aligned_right.exogenous) .* 2
+    combined_matrix = left_matrix .+ right_matrix
+    combined_mvts = MVTSeries(first(rangeof(left)), collect(keys(left.varshks)), combined_matrix)
+    combined_mvts = combined_mvts[intersect(rangeof(left),rangeof(right)), intersect(left_vars, right_vars)]
+
+    
+
+    if summary || diff
+        differences = combined_mvts .% 3 .== 0
     end
-)
-function compare_plans(io::IO, left::Plan, right::Plan;
-    alphabetical=false,  # whether to sort variables
-    pagelines=0,
-    exog_mark="X", endo_mark="~", missing_mark=".", # symbols used for each class
-    delim=" ",  # set to "," to get a CSV file (with 3 skip rows and 1 header row)
-    _name_delim=delim,  # padding after NAME column
-    _range_delim=delim    # padding between range columns
-)
-    # summary(io, p)
+    if summary
+        if left.range == right.range
+            println("Same range: ", left.range)
+        else
+            println("Range  left: ", left.range)
+            println("Range right: ", right.range)
+        end
+        
+        if Set(left_vars) == Set(right_vars)
+            println("Same variables.")
+        else
+            println("Variables only in left plan: ", setdiff(left_vars, right_vars))
+            println("Variables only in right plan: ", setdiff(right_vars, left_vars))
+            println(length(intersect(left_vars, right_vars)), " common variables.")
+        end
+        for var in keys(combined_mvts)
+            mits = rangeof(combined_mvts[var])[differences[var] .== false]
+            if length(mits) > 0
+                ranges = _mits_to_ranges(mits)
+                str = "$var differs between the plans for the range(s) $(ranges[1])"
+                for rng in ranges[2:end]
+                    str *= ", $rng"
+                end
+                println("$str.")
+            end
```

Comment:
Why not `print()` each piece directly, instead of building up the whole string and printing it all at once.  I think printing each chunk is better since it doesn't require all the memory management of string manipulations.  (I assume the io stream would have a pre-allocated buffer, so writing to it would be cheap, but I don't actually know)

Reply:
I just ran a simple test to check this.  `*=`  uses more time and memory than `print`. 

![image](https://github.com/bankofcanada/StateSpaceEcon.jl/assets/12414774/d0fd7071-64b1-4989-a987-8477ded8cce7)

Reply:
Thanks for checking that. Fixed!